spec/plaindate.html

```diff
@@ -954,6 +954,38 @@ <h1>AddISODate ( _year_, _month_, _day_, _years_, _months_, _weeks_, _days_, _ov
       </emu-alg>
     </emu-clause>
 
+    <emu-clause id="sec-temporal-adddate" type="abstract operation">
+      <h1>
+        AddDate (
+          _calendar_: an Object,
+          _plainDate_: a Temporal.PlainDate,
+          _duration_: a Temporal.Duration,
+          optional _options_: an Object or *undefined*,
+          optional _dateAdd_: a function Object or ~unused~,
+        ): either a normal completion containing a Temporal.PlainDate, or an abrupt completion
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>
+          It adds _duration_ to _plainDate_.
+          Usually this is accomplished by calling the `dateAdd` method of the given _calendar_ as in CalendarDateAdd, but if the duration only contains days, the computation will not cause an observable method call.
+          If _dateAdd_ is present, the `dateAdd` method will not be observably looked up.
+        </dd>
+      </dl>
+      <emu-alg>
+        1. If _options_ is not present, set _options_ to *undefined*.
+        1. If _duration_.[[Years]] &ne; 0, or _duration_.[[Months]] &ne; 0, or _duration_.[[Weeks]] &ne; 0, then
+          1. If _dateAdd_ is not present, then
+            1. Set _dateAdd_ to ~unused~.
+            1. If _calendar_ is an Object, set _dateAdd_ to ? GetMethod(_calendar_, *"dateAdd"*).
```

Comment:
This is because (as far as I know) there's no convenient way in spec syntax to perform CalendarDateAdd(_param1, param2, param3, param4_) if _dateAdd_ is absent, and CalendarDateAdd(_param1, param2, param3, param4, dateAdd_) if _dateAdd_ is present. It would be an editorial error to unconditionally call CalendarDateAdd(_param1, param2, param3, param4, dateAdd_) if _dateAdd_ is absent.

I could probably find a nicer solution for this if I invested more time, but the _dateAdd_ parameter is going away anyway in Part 4 (#2519), to be replaced by a Calendar Methods Record. So I'd prefer to leave it as is for now.

Reply:
Ah, so editorial shenanigans, not user code call shenanigans. Alright, thanks for clearing that up, I can't see a need for any changes!